src/hotspot/share/gc/shared/space.cpp

```diff
@@ -140,11 +140,6 @@ void ContiguousSpace::verify() const {
   guarantee(p == top(), "end of last object must match end of space");
 }
 
-bool Space::obj_is_alive(const HeapWord* p) const {
-  assert (block_is_obj(p), "The address should point to an object");
```

Comment:
Does this remove the last usage of block_is_obj? If it does, that function should probably also be removed in this function.